openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1847,10 +1854,16 @@ TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterialServerSide)(TCN_STDARGS, jlong ssl,
 
 TCN_IMPLEMENT_CALL(void, SSL, setKeyMaterialClientSide)(TCN_STDARGS, jlong ssl, jlong certOut, jlong keyOut, jlong chain, jlong key)
 {
-#if OPENSSL_VERSION_NUMBER < 0x10002000L || defined(LIBRESSL_VERSION_NUMBER)
-    // Only supported on openssl 1.0.2+
-    tcn_Throw(e, "Not supported");
+#if defined(LIBRESSL_VERSION_NUMBER)
+    tcn_Throw(e, "Not supported with LibreSSL");
 #else
+#ifndef OPENSSL_IS_BORINGSSL
```

Comment:
this is somewhat confusing.  You might consider putting in a comment which `endif` is associated.  e.g.

```
#ifndef OPENSSL_IS_BORINGSSL
...
#endif // OPENSSL_IS_BORINGSSL
```

Reply:
done